x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -335,15 +337,7 @@ private void processValidatedJwt(
         }
 
         // User metadata: If enabled, extract metadata from JWT claims set. Use it in UserRoleMapper.UserData and User constructors.
-        final Map<String, Object> userMetadata;
-        try {
-            userMetadata = populateUserMetadata ? JwtUtil.toUserMetadata(claimsSet) : Map.of();
-        } catch (Exception e) {
-            final String msg = "Realm [" + name() + "] parse metadata failed for principal=[" + principal + "].";
-            logger.debug(msg, e);
-            listener.onResponse(AuthenticationResult.unsuccessful(msg, e));
-            return;
-        }
+        final Map<String, Object> userMetadata = buildUserMetadata(claimsSet);
```

Comment:
I removed the error handling here because I don't think we should convert an exception to unsuccessful authentication (and fall through to the next realm) in this case. Falling through the realm chain should be used with care, its intention is not to hide potential exceptions. It is useful mainly when the incoming credentials fail to be verified against the stored one. But any other exceptions, especially ones after credentials are validated (like this one here), should be reported as failure. 

In practice, there is hardly any exception that can be thrown in the metadata population process. So the change is mostly not visible to users (if at all).